packages/adapter-components/test/elements/swagger/instance_elements.test.ts

```diff
@@ -1029,5 +1029,139 @@ describe('swagger_instance_elements', () => {
         objectTypes,
       })).rejects.toThrow(new Error('Invalid type config - type myAdapter.Pet has no request config'))
     })
+    describe('element ids and file names transformation', async () => {
+      const NAME = 'name'
+      const ID = 'id'
+      const NUMBER = 9
+      interface GetInstanceParams {
+        fileNameFields?: string[]
+        idFields?: string[]
+      }
+      const getInstance = async ({ fileNameFields = [], idFields = [] }
+                                     : GetInstanceParams): Promise<InstanceElement> => {
+        const paginator: Paginator = jest.fn((
+          async function *getAll(_, extractPageEntries) {
+            yield [
+              {
+                id: ID,
+                name: NAME,
+                number: NUMBER,
+                list: [1, 2, 3],
+              },
+            ].flatMap(extractPageEntries)
+          }))
+        const customType = new ObjectType({
+          elemID: new ElemID(ADAPTER_NAME, 'customType'),
+          fields: {
+            name: { refType: BuiltinTypes.STRING },
+            id: { refType: BuiltinTypes.STRING },
+            number: { refType: BuiltinTypes.NUMBER },
+          },
+        })
+        return (await getAllInstances({
+          paginator,
+          apiConfig: {
+            typeDefaults: {
+              transformation: {
+                idFields: ['id'],
+              },
+            },
+            types: {
+              customType: {
+                request: {
+                  url: '/someObject',
+                },
+                transformation: {
+                  fileNameFields,
+                  idFields,
+                },
+              },
+            },
+          },
+          fetchConfig: {
+            includeTypes: ['customType'],
+          },
+          objectTypes: { customType },
+          computeGetArgs: simpleGetArgs,
+          nestedFieldFinder: returnFullEntry,
+        }))[0]
+      }
+
+      describe('file name', async () => {
+        test('generates name with non missing fields', async () => {
+          const instance = await getInstance({ fileNameFields: ['id', 'name', 'number'] })
+          const { path } = instance
```

Comment:
I think this code remove some of the boiler plate code.
```suggestion
          const filename = instancePathFilename(await getInstance({ fileNameFields: ['id', 'name', 'number'] }))
```

Reply:
v